ratis-examples/src/main/java/org/apache/ratis/examples/common/Common.java

```diff
@@ -16,35 +16,55 @@
  * limitations under the License.
  */
 
-package org.apache.ratis.examples.counter;
+package org.apache.ratis.examples.common;
 
 import org.apache.ratis.protocol.RaftGroup;
 import org.apache.ratis.protocol.RaftGroupId;
 import org.apache.ratis.protocol.RaftPeer;
 
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Properties;
 import java.util.UUID;
 
 /**
  * Common Constant across servers and client
  */
-public final class CounterCommon {
+public final class Common {
   public static final List<RaftPeer> PEERS;
 
   static {
     List<RaftPeer> peers = new ArrayList<>(3);
-    peers.add(RaftPeer.newBuilder().setId("n1").setAddress("127.0.0.1:6000").build());
-    peers.add(RaftPeer.newBuilder().setId("n2").setAddress("127.0.0.1:6001").build());
-    peers.add(RaftPeer.newBuilder().setId("n3").setAddress("127.0.0.1:6002").build());
+    Properties properties = new Properties();
+
+    try(InputStream inputStream = new FileInputStream("ratis-examples/src/main/resources/conf.properties");
+        Reader reader = new InputStreamReader(inputStream, "UTF-8");
+        BufferedReader bufferedReader = new BufferedReader(reader)) {
+      properties.load(bufferedReader);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    String[] address = properties.getProperty("raft.server.address").split(",");
+    if (address.length <= 0) {
+      throw new IllegalArgumentException("Can't get the ratis server address");
+    }
+    for (int i = 0; i < address.length; i++) {
+      peers.add(RaftPeer.newBuilder().setId("n" + i).setAddress(address[i]).build());
+    }
```

Comment:
- Let's print out the conf file and the key in the error messages.
- Use Optional to handle null.
```
  static {
    final Properties properties = new Properties();
    final String conf = "ratis-examples/src/main/resources/conf.properties";
    try(InputStream inputStream = new FileInputStream(conf);
        Reader reader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
        BufferedReader bufferedReader = new BufferedReader(reader)) {
      properties.load(bufferedReader);
    } catch (IOException e) {
      throw new IllegalStateException("Failed to load " + conf, e);
    }
    final String key = "raft.server.address.list";
    final String[] addresses = Optional.ofNullable(properties.getProperty(key))
        .map(s -> s.split(","))
        .orElse(null);
    if (addresses == null || addresses.length == 0) {
      throw new IllegalArgumentException("Failed to get " + key + " from " + conf);
    }

    final List<RaftPeer> peers = new ArrayList<>(addresses.length);
    for (int i = 0; i < addresses.length; i++) {
      peers.add(RaftPeer.newBuilder().setId("n" + i).setAddress(addresses[i]).build());
    }
    PEERS = Collections.unmodifiableList(peers);
  }
```

Reply:
done